src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,145 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.*;
+import duke.storage.Storage;
+import duke.task.*;
+import duke.ui.Ui;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Encapsulates the processing of user inputs to duke.
+ *
+ * @author Zhi Bin
+ * @version Duke Level 9
+ */
+public class Parser {
+    private final TaskList list;
+    private final Storage storage;
+    private final Ui ui;
+
+    private enum TaskTypes {
+        TODO, DEADLINE, EVENT
+    }
+
+    /**
+     * Creates a Parser that handles user input and turn it into respective
+     * commands and execute the required functions that user demands.
+     *
+     * @param list    The TaskList handler that is handling the list of task.
+     * @param storage The Storage handler that is in-charged of saving and loading files on local directory.
+     * @param ui      The Ui handler that handles printing of output, if required.
+     */
+    public Parser(TaskList list, Storage storage, Ui ui) {
+        this.list = list;
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    /**
+     * Returns the respective DukeCommand that user want duke to do.
+     * If the command is not supported
+     *
+     * @param userInput User's input on what they want duke to do.
+     * @return The respective command object.
+     * @throws DukeException If some error occurred in the processing of user input,
+     *                       like the user inputted some command that duke does not support.
+     */
+    public DukeCommand processInput(String userInput) throws DukeException {
```

Comment:
is it possible to have different methods to process the different inputs? might make it look less messy

Reply:
hmmm you mean like one method for each case? But the methods will be like one-liner though

Reply:
I think if you want to make it more convenient to receive multiple arguments you can use varargs. for the process input method